File: Source/Extensions/Tessa.Extensions.Default.Client/UI/KrProcess/KrStageSourceUIExtension.cs
using System.Threading.Tasks;
using Tessa.Extensions.Default.Client.Workflow.KrProcess;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls;

namespace Tessa.Extensions.Default.Client.UI.KrProcess
{
    /// <summary>
    /// Расширение на кнопки в карточках шаблона этапа KrStageTemplates и базового метода KrCommonMethod
    /// </summary>
    public sealed class KrStageSourceUIExtension: CardUIExtension
    {
        #region constants

        private const string CompileButtonAlias = "CompileButton";

        private const string CompileAllButtonAlias = "CompileAllButton";

        #endregion
        
        #region base overrides

        public override async Task Initialized(ICardUIExtensionContext context)
        {
            var cardModel = context.Model;

            if (cardModel.Controls.TryGet(CompileButtonAlias, out var compileControl)
                && compileControl is ButtonViewModel compileButton)
            {
                compileButton.CommandClosure.Execute = async o => await KrUIHelper.SendCompileRequestAsync(KrConstants.Keys.CompileWithValidationResult);
            }

            if (cardModel.Controls.TryGet(CompileAllButtonAlias, out var compileAllControl) 
                && compileAllControl is ButtonViewModel compileAllButton)
            {
                compileAllButton.CommandClosure.Execute = async o => await KrUIHelper.SendCompileRequestAsync(KrConstants.Keys.CompileAllWithValidationResult);
            }
        }

        #endregion
    }
}

File: WebClient SDK/src/default/ui/krProcess/krStageSourceUIExtension.ts
import { CompiledCardTypes, sendCompileRequest } from '../../workflow/krProcess/krUIHelper';
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { ButtonViewModel } from 'tessa/ui/cards/controls';

export class KrStageSourceUIExtension extends CardUIExtension {

  public initialized(context: ICardUIExtensionContext) {
    if (!CompiledCardTypes.some(x => x === context.card.typeId)) {
      return;
    }

    const cardModel = context.model;

    const compileControl = cardModel.controls.get('CompileButton') as ButtonViewModel;
    if (compileControl) {
      compileControl.onClick = () => sendCompileRequest('.CompileWithValidationResult');
    }

    const compileAllControl = cardModel.controls.get('CompileAllButton') as ButtonViewModel;
    if (compileAllControl) {
      compileAllControl.onClick = () => sendCompileRequest('.CompileAllWithValidationResult');
    }
  }

}